cmd/serf/command/agent/config.go

```diff
@@ -344,6 +349,21 @@ func DecodeConfig(r io.Reader) (*Config, error) {
 		result.BroadcastTimeout = dur
 	}
 
+	if result.NodeName != "" && result.ValidateNodeNames {
+		var InvalidNameRe = regexp.MustCompile(`[^A-Za-z0-9\\-]+`)
+		if InvalidNameRe.MatchString(result.NodeName) {
+			err = fmt.Errorf("NodeName contains invalid characters %v , Valid characters include "+
+				"all alpha-numerics and dashes.", result.NodeName)
+			return nil, err
+		}
+
+		if len(result.NodeName) > serf.MaxNodeNameLength {
+			err = fmt.Errorf("NodeName is %v characters. "+
+				"Valid length is between 1 and 128 characters", len(result.NodeName))
+			return nil, err
+		}
```

Comment:
This logic for validating the name is getting repeated 3 times - we should pull it into a function for reuse. We can also avoid doing it in this `serf/command/agent` code at all and do it here instead: https://github.com/hashicorp/serf/blob/master/serf/serf.go#L235

`serf.Create` will get called on startup for Consul as well, where the code in this package is only used in the Serf binary.